From 7b1a13228b321ea27bd53070bcd332417069ace8 Mon Sep 17 00:00:00 2001
From: Nan Li <nli@suse.com>
Date: Wed, 4 Mar 2015 18:48:35 +0800
Subject: [PATCH] ACPI / sysfs: Treat the count field of counter_show() as
 unsigned

Git-commit: 7b1a13228b321ea27bd53070bcd332417069ace8
Patch-mainline: v4.1
References: bnc#909312
Target: SLE-11 SP3

The count field is an unsigned 32bit value, and the
counter_show() function should also treat it as a unsigned
value.

Otherwise the counter may show negative number as we found on a
machine:
...
gpe23:        0   invalid
gpe24: -2071733   enabled
gpe25:        0   invalid
...
gpe_all: -2070980
sci: -2070949

Signed-off-by: Nan Li <nli@suse.com>
Signed-off-by: Lee, Chun-Yi <jlee@suse.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---
 drivers/acpi/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-3.0-SLE11-SP3/drivers/acpi/sysfs.c
===================================================================
--- linux-3.0-SLE11-SP3.orig/drivers/acpi/sysfs.c
+++ linux-3.0-SLE11-SP3/drivers/acpi/sysfs.c
@@ -524,7 +524,7 @@ static ssize_t counter_show(struct kobje
 	    acpi_irq_not_handled;
 	all_counters[num_gpes + ACPI_NUM_FIXED_EVENTS + COUNT_GPE].count =
 	    acpi_gpe_count;
-	size = sprintf(buf, "%8d", all_counters[index].count);
+	size = sprintf(buf, "%8u", all_counters[index].count);
 
 	/* "gpe_all" or "sci" */
 	if (index >= num_gpes + ACPI_NUM_FIXED_EVENTS)
